### PR TITLE
MEN-4475: port-forwarding integration test using mender-cli

### PR DESCRIPTION
**Pull request by tranchitella**
Changelog: none

Signed-off-by: Fabio Tranchitella <fabio.tranchitella@northern.tech>

### CODE REVIEW

**Review by lluiscampos — COMMENTED** [review]
A part from the `mender-qa` required changes (PR coming up in few minutes), we need to add it in the `run.sh` wrapper on this repository too for local runs. Something similar to [these lines](https://github.com/mendersoftware/integration/blob/master/tests/run.sh#L89-L96).

File: tests/tests/test_portforward.py [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+        )
+        p.start()
+
+        # wait a few seconds to let the port-forward start
+        logger.info("port-forward started, waiting a few seconds")
+        time.sleep(2)
+
+        #  verify the UDP port-forward querying the Google's DNS server
+        logger.info("resolve mender.io (A record)")
+        req = DnsRequest(name="mender.io", qtype="A", server="localhost", port=8853)
```
Comment (kacf) [inline tests/tests/test_portforward.py:105]: Smart!

File: tests/tests/test_portforward.py [hunk 2]
```diff
@@ -0,0 +1,173 @@
...
+                ]
+                + list(port_mappings),
+                stdout=subprocess.PIPE,
+            )
+            stdout, stderr = p.communicate()
+            p.wait()
+
+        p = Process(
+            target=port_forward,
+            args=(server_url, devid, "8822:22", "udp/8853:8.8.8.8:53"),
```
Comment (kacf) [inline tests/tests/test_portforward.py:95]: Just to reduce the risk of confusion: Could we pick a different port than 8822? This port is used as the default ssh port into QEMU instances; using a different port number would make it clearer that you are not taking that route here (connecting to `localhost`, not the QEMU IP).

**Review by kacf — APPROVED** [review]
One suggestion to make the intention a bit clearer, but looks good overall!

**Review by lluiscampos — CHANGES_REQUESTED** [review]
The test looks very good, nice job.

> A part from the `mender-qa` required changes (PR coming up in few minutes), we need to add it in the `run.sh` wrapper on this repository too for local runs. Something similar to [these lines](https://github.com/mendersoftware/integration/blob/master/tests/run.sh#L89-L96).

Requesting changes for this ^

**Comment by tranchitella** [comment]
@lluiscampos done: https://github.com/mendersoftware/integration/pull/1272/files#diff-dd0c9cda3917fccebea6cea8d3fb46f46e4fbbeda2d6939864c9aaa9fc9c9571R99

File: tests/run.sh [hunk 3]
```diff
@@ -90,6 +96,10 @@ function get_requirements() {
          -o downloaded-tools/mender-artifact \
          -z downloaded-tools/mender-artifact
 
+    curl --fail "https://downloads.mender.io/mender-cli/${MENDER_CLI_BRANCH}/linux/mender-cli" \
+         -o downloaded-tools/mender-artifact \
+         -z downloaded-tools/mender-artifact
```
Comment (lluiscampos) [inline tests/run.sh:101]: typo

```suggestion
    curl --fail "https://downloads.mender.io/mender-cli/${MENDER_CLI_BRANCH}/linux/mender-cli" \
         -o downloaded-tools/mender-cli \
         -z downloaded-tools/mender-cli
```

**Review by lluiscampos — APPROVED** [review]
Typo, see below. Pre-approving :heavy_check_mark:

**Comment by lluiscampos** [comment]
FYI https://github.com/mendersoftware/mender-qa/pull/494 is working now. If you need to verify the test here before `mender-qa` PR is merged, specify it when launching the build and it should be fine.

**Comment by tranchitella** [comment]
Pipeline: https://gitlab.com/Northern.tech/Mender/mender-qa/-/pipelines/274332946

**Pull request merged**